src/service/command/execute-command-service.ts

```diff
@@ -39,10 +39,18 @@ export class ExecuteCommandService {
     const before = this.getNodeCommands(node, ExecutionPhase.BEFORE, this._configurationService.getNodeExecutionLevel(node));
     const current = this.getNodeCommands(node, ExecutionPhase.CURRENT, this._configurationService.getNodeExecutionLevel(node));
     const after = this.getNodeCommands(node, ExecutionPhase.AFTER, this._configurationService.getNodeExecutionLevel(node));
-    const skipExecution = this._configurationService.skipExecution(node);
-    result.push(await this.executeCommands(node, before ?? [], skipExecution, {...opts, cwd}));
-    result.push(await this.executeCommands(node, current ?? [], skipExecution, {...opts, cwd}));
-    result.push(await this.executeCommands(node, after ?? [], skipExecution, {...opts, cwd}));
+    let skipExecution = this._configurationService.skipExecution(node);
+
+    if (!(await this.executeNodePhase(node, before, skipExecution, {...opts, cwd}, result))) {
+      skipExecution = true;
```

Comment:
why updating `skipExecution` ?

Can't we just return the result if this step fails ?

Reply:
After each projects execution a result array is returned:
```typescript
[
    {
        node,
        results: [...] // before results  
    },
    {
        node,
        results: [...] // current results  
    },
    {
        node,
        results: [...] // after results  
    },
]
```
Even when a phase doesn't have defined commands, this array has 3 elements and corresponding phase's result array is just simply empty.

So to maintain same data structure I updated `skipExecution` instead of returning just so that I can guarantee 3 elements in that array. I could have alternatively just returned an empty result arrays for the other sections but the job summary will translate that as we had defined no commands for the section which might be misleading.

Reply:
I could but I won't be able to combine them all. So it would look like this:
```typescript
let skipExecution = this._configurationService.skipExecution(node);
skipExecution ||= !(await this.executeNodePhase(node, before, skipExecution, {...opts, cwd}, result))
skipExecution ||=  !(await this.executeNodePhase(node, current, skipExecution, {...opts, cwd}, result))
```
Is this okay?

Reply:
This is what I meant :)

Reply:
For some reason this doesn't wait for the promise to resolve, so I had to reverse the order which worked somehow
```typescript
let skipExecution = this._configurationService.skipExecution(node);
skipExecution = !(await this.executeNodePhase(node, before, skipExecution, {...opts, cwd}, result)) || skipExecution;
skipExecution = !(await this.executeNodePhase(node, current, skipExecution, {...opts, cwd}, result)) || skipExecution;
```